classes/solid/terrain/pencil.gd

```diff
@@ -273,23 +253,24 @@ func get_connected_lines_directional(lines, add_onto_blacklist, direction, poly,
 		var normal: Vector2 = vert.direction_to(next).tangent()
 		var angle: float = normal.angle_to(direction) / PI * 180
 		
-		if angle >= -root.max_deviation and angle <= root.max_deviation:
+		if (!type_list.has(ind) and angle >= -root.max_deviation and angle <= root.max_deviation) or (type_list.has(ind) and type_list[ind] == type_id):
```

Comment:
Possibly make a seperate function which returns a bool if all requirements are met, makes this part of the code more readable